### PR TITLE
Enable tests for CPU tensors in test_distributed.py

### LINKED ISSUE #14536: How to use torch.distributed.gather?
I compiled pytorch(1.0.0a0+ff608a9) with openMPI. When I tried to use the `torch.distributed.gather`, I found `gather_list` is required. But I think as sender it seems not necessary to recv `list[Tensor]`. So I just set `gather_list` to `None`. Well, when the script is running, the error comes:
 
> File "test.py", line 216, in train
    dist.gather(outputs, gather_list=None, dst=0, group=new_group)
  File "./lib/python2.7/site-packages/torch/distributed/distributed_c10d.py", line 986, in gather
    work = group.gather([gather_list], [tensor], opts)
TypeError: gather(): incompatible function arguments. The following argument types are supported:
    1. (self: torch.distributed.ProcessGroup, arg0: List[List[at::Tensor]], arg1: List[at::Tensor], arg2: torch.distributed.GatherOptions) -> c10d::ProcessGroup::Work
    2. (self: torch.distributed.ProcessGroup, output_tensors: List[at::Tensor], tensor: at::Tensor, root: int) -> c10d::ProcessGroup::Work
Invoked with: <torch.distributed.ProcessGroupMPI object at 0x7f63064b1a40>, [None], [tensor([[9.2869e-01, 8.5623e-01, 1.0708e+00,  ...

So how can I solve this error? As there is few documents about how to use collective operations, I wonder if there is a wiki or website because current tutorial is outdated.

Thank you!

### PR DESCRIPTION
**Pull request by pietern**
These were not enabled after adding support in the Gloo backend. The
argument checks in ProcessGroupGloo raised an error in two cases:

* If the input tensor list to scatter was ``[None]`` on processes other
  than the source process.
* If the output tensor list to gather was ``[None]`` on processes other
  than the destination process.

This commit prepares these arguments explicitly instead of boxing them
at the process group call site.

This fixes #14536.

### CODE REVIEW

File: torch/lib/c10d/ProcessGroupMPI.cpp
Comment (teng-li) [inline torch/lib/c10d/ProcessGroupMPI.cpp]: Why do you need this check? checkSingleTensor has done this already in line 478

File: torch/lib/c10d/ProcessGroupMPI.cpp
Comment (teng-li) [inline torch/lib/c10d/ProcessGroupMPI.cpp]: You don't need this as well

Comment (pietern) [reply torch/lib/c10d/ProcessGroupMPI.cpp]: Removing.

File: torch/lib/c10d/ProcessGroupMPI.cpp
Comment (pietern) [reply torch/lib/c10d/ProcessGroupMPI.cpp]: Removing.

**Comment by teng-li** [comment]
wait for the CI before landing I guess

**Review by teng-li — APPROVED** [review]

**Pull request closed without merging**
